core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1009,8 +1006,8 @@ private DeleteResult deleteInLucene(Delete delete, DeletionStrategy plan)
         } catch (Exception ex) {
             if (indexWriter.getTragicException() == null) {
                 // there is no tragic event and such it must be a document level failure
-                return new DeleteResult(ex, plan.versionOfDeletion, plan.versionOfDeletion,
-                    plan.currentlyDeleted == false);
+                return new DeleteResult(
+                        ex, plan.versionOfDeletion, plan.seqNoOfDeletion, plan.currentlyDeleted == false);
```

Comment:
@bleskes By the way, I think this was a bug, maybe there's a test case missing here?